File: taskmanager/src/main/java/com/capusule/fse/taskmanager/models/Task.java
package com.capusule.fse.taskmanager.models;


import org.springframework.lang.NonNull;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@ToString
@NoArgsConstructor
@JsonInclude(Include.NON_NULL)
public class Task {

	@NonNull
	private String task;
	
	private String parentTask;
	
	@NonNull
	private int priority;
	
	@NonNull
	private String startDate;
	
	@NonNull
	private String endDate;
	
	private String taskId;
	
	private String parentTaskId;
	
	private boolean editable;
}

File: taskmanagerUI/src/app/task/model/task.ts
export class Task {
    task:string;
    parentTask:string;
    priority:number;
    startDate:string;
    endDate:string;
    taskId:number;
    parentTaskId:number;
    editable:boolean;

    constructor(){
        this.task='';
        this.priority=15;
        this.startDate='';
        this.endDate='';
    }
}